app/objects/c_agent.py

```diff
@@ -84,6 +85,8 @@ def store(self, ram):
         self.last_seen = now
         if self.trusted:
             self.last_trusted_seen = now
+        if kwargs.get('error'):
```

Comment:
i'm not clear what this does?

Reply:
The "error" property is sent by the agent in the event that it could not successfully switch communication channels.  If the error is present the agent will switch to the default communication channel for that particular agent.